### DISCUSSION TITLE
WebSocket connection to 'wss://...' failed:

### DISCUSSION THREAD

**Discussion by marc-odp** [zauberzeug/nicegui#2689] [Q&A]
### Question

I have a button on my main page with "ui.navigate.to('/login')"
When I use the LAN ip address of the server, I can click on the button and I see the login page.
When I use the domain name address (with https and reverse proxy), clicking on the button do nothing but I see in the developer tools of the browser :
`WebSocket connection to 'wss://...' failed: `
What am I doing wrong ?

**Comment by rodja** [comment]
Likely your reverse proxy is not correctly forwarding the websocket traffic. See https://github.com/zauberzeug/nicegui/tree/main/examples/nginx_subpath for an Nginx example. And https://github.com/zauberzeug/nicegui/blob/main/docker-compose.yml for a Traefik configuration.

**Reply by marc-odp** [reply]
I am using the Synology NAS reverse proxy server feature. 
I found out that if I go to the "Custom Header" tab and click create, two values are automatically created. When I save this, the problem is gone.
I have no idea what those extra settings are doing. I have other docker containers on this NAS using reverse proxy and I never had to create custom header settings for them...
Thank you for your help !

![image](https://github.com/zauberzeug/nicegui/assets/17361839/64a58db1-b84f-451b-ac19-faa3e15d3ee2)

**Reply by rodja** [reply]
Looks good. The headers are needed for websockets. The initial connection is made via http and then "upgraded" to websocket.

**Comment by marc-odp** [comment]
Thanks !
